lib/zipkin-tracer/zipkin_tracer_base.rb

```diff
@@ -23,7 +23,16 @@ def with_new_span(trace_id, name)
 
     def end_span(span)
       span.close
-      if span.annotations.any?{ |ann| ann.value == Annotation::SERVER_SEND }
+      # if in a thread not handling incoming http requests, it will not have Annotation::SERVER_SEND, so the span
+      # will never be flushed and will cause memory leak.
+      # It will have CLIENT_SEND and CLIENT_RECV if the thread sends out http requests, so use CLIENT_RECV as the sign
```

Comment:
Maybe is a cornercase but zipkin allows for local spans. Yo could for instance trace calls to DBs or local programs or anything that is interesting to trace without even calling an external service. I guess this at least solves some uses cases.

Another comment is that spans access a thread local variable which is slow, this change would mean an extra access each span which would make tracing slower for everyone